### PR TITLE
Corporate Portal configuration

### PR DESCRIPTION
**Pull request by MagdalenaZuba**
| Question      | Answer
| ------------- | ---
| JIRA Ticket   | N/A
| Versions      | 4.3

#### Checklist

- [ ] Text renders correctly
- [ ] Text has been checked with vale
- [ ] Description metadata is up to date
- [ ] Redirects cover removed/moved pages
- [ ] Code samples are working
- [ ] PHP code samples have been fixed with PHP CS fixer
- [ ] Added link to this PR in relevant JIRA ticket or code PR

### CODE REVIEW

File: docs/customer_management/cp_configuration.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+## `corporate` SiteAccess
+
+The predefined `corporate` SiteAccess in `corporate_group`
+(configured in `config/packages/ibexa.yaml`) serves the Customer Portal.
+If you need a multisite setup with multiple Customer Portals,
+add any additional SiteAccesses to `corporate_group`.
+
+## Customer identifier
+
+`ibexa_default_settings.yaml` contains setting that indicate which Content Types are treated like customers:
```
Comment (ViniTou) [inline docs/customer_management/cp_configuration.md:20]: This is not true, that settings just say what content types should be treated like `Users` in terms for example usage in `UserService`.
There is no way to assign which CT should be treated like `cutomer`, as there is no specific behavior for that.

File: docs/customer_management/cp_configuration.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+parameters:
+    ibexa.site_access.config.default.corporate_accounts.roles:
+        admin: Company Admin
+        buyer: Company Buyer
+        custom_role: Company Assistant
+```
+
+## Content Types names
+
+To change names of default Content Types, you need to define them in the configuration
```
Comment (ViniTou) [inline docs/customer_management/cp_configuration.md:49]: That doesnt make sense, name changing is done in UI, what you could do is assign what content types should be used to describe `Company` and `Member` in Domain layer.
But those configs are done under `ibexa_corporate_account` key, like shown in `\Ibexa\Bundle\CorporateAccount\DependencyInjection\Configuration`.

So if you already have `company` CT in your system, and you dont want to change that identifier, then you should configure CA in following way.
```
ibexa_corporate_account:
    content_type_mappings:
        company: your_ct_identifier
```

Be aware tho, that you have to adjust migration files by yourself if you are in that position.

File: docs/customer_management/cp_configuration.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+## `corporate` SiteAccess
+
+The predefined `corporate` SiteAccess in `corporate_group`
+(configured in `config/packages/ibexa.yaml`) serves the Customer Portal.
+If you need a multisite setup with multiple Customer Portals,
+add any additional SiteAccesses to `corporate_group`.
+
+## Customer identifier
+
+`ibexa_default_settings.yaml` contains setting that indicate what Content Types should be treated like Users in terms for example usage in `UserService`:
```
Comment (ViniTou) [inline docs/customer_management/cp_configuration.md:20]: Not sure however, how this relates to customer portal, and it is already mentioned in https://doc.ibexa.co/en/latest/administration/configuration/repository_configuration/#user-identifiers

**Review by ViniTou — APPROVED** [review]

**Review by Nattfarinn — APPROVED** [review]

File: docs/customer_management/cp_configuration.md (hunk 3 above)
Comment (DominikaK) [inline docs/customer_management/cp_configuration.md:20]:
```suggestion
`ibexa_default_settings.yaml` contains a setting that indicates what Content Types should be treated like Users in terms of, for example, usage in `UserService`:
```

File: docs/customer_management/cp_configuration.md [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+ibexa:
+    system:
+        default:
+            user_content_type_identifier: ['user', 'customer']
+
+```
+
+## Roles and Policies
+
+You can easily add custom roles to your installation
```
Comment (DominikaK) [inline docs/customer_management/cp_configuration.md:32]:
```suggestion
You can add custom roles to your installation
```
"Never say simple" ;)

File: docs/customer_management/cp_configuration.md [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    ibexa.site_access.config.default.corporate_accounts.roles:
+        admin: Company Admin
+        buyer: Company Buyer
+        custom_role: Company Assistant
+```
+
+## Content Types names
+
+You can change names of default Content Types by assigning what
+Content Types should be used to describe `Company` and `Member` in domain layer.
```
Comment (DominikaK) [inline docs/customer_management/cp_configuration.md:46]:
```suggestion
Content Types should be used to describe `Company` and `Member` in the Back Office.
```
maybe, if that's true?

File: docs/customer_management/cp_configuration.md [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+Proceed only if you already have `Company` Content Type in your system, and you don't want to change its identifier.
```
Comment (DominikaK) [inline docs/customer_management/cp_configuration.md:47]:
```suggestion
Proceed only if you already have a `Company` Content Type in your system, and you don't want to change its identifier.
```

File: docs/customer_management/cp_configuration.md [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+Configuration for Content Type names is placed under `ibexa_corporate_account` key,
```
Comment (DominikaK) [inline docs/customer_management/cp_configuration.md:49]:
```suggestion
Configuration for Content Type names is placed under the `ibexa_corporate_account` key,
```

File: docs/customer_management/cp_configuration.md [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+like shown in `Ibexa\Bundle\CorporateAccount\DependencyInjection\Configuration`.
+To change Content Types names, adjust corporate account configuration in the following way:
```
Comment (DominikaK) [inline docs/customer_management/cp_configuration.md:51]:
```suggestion
To change Content Type names, adjust corporate account configuration in the following way:
```

File: docs/customer_management/cp_configuration.md [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+```yaml
+parameters:
+    ibexa.site_access.config.default.corporate_accounts.roles:
+        admin: Company Admin
+        buyer: Company Buyer
+        custom_role: Company Assistant
+```
+
+## Content Types names
```
Comment (DominikaK) [inline docs/customer_management/cp_configuration.md:43]:
```suggestion
## Content Type names
```

File: docs/customer_management/cp_configuration.md [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+        - first_name
+        - last_name
+        - phone_number
+        - user
+```
+
+## Address
+
+With the Address Field Type, you can customize address Fields and configure them per country.
+To lern more see [Address Field Type documentation](../content_management/field_types/field_type_reference/addressfield.md).
```
Comment (DominikaK) [inline docs/customer_management/cp_configuration.md:84]:
```suggestion
To learn more, see [Address Field Type documentation](addressfield.md).
```

File: docs/customer_management/cp_configuration.md [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+
+## Templates
+
+You can also define new templates for among others: invitation email,
```
Comment (DominikaK) [inline docs/customer_management/cp_configuration.md:88]:
```suggestion
You can also define new templates for, among others: invitation email,
```

File: docs/customer_management/create_user_registration_form.md [hunk 12]
```diff
@@ -7,13 +7,13 @@ description: Customize the registration form for new users in your site front en
 You can create a registration form for users to your website by creating a Twig template or editing the existing registration form in YAML file.
 Follow the instructions below to create and customize templates for a registration form, and a registration confirmation page.
 
-First, make sure you [enabled user registration](permission_use_cases.md#register-users).
+First, make sure you [enabled user registration](../permissions/permission_use_cases.md#register-users).
```
Comment (DominikaK) [inline docs/customer_management/create_user_registration_form.md:10]: Is this change needed? Autolinks should cover it.

File: docs/customer_management/customer_portal.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+# Customer Portal
+
+Customer Portal enables [creating and handling of business accounts](https://doc.ibexa.co/projects/userguide/en/latest/shop_administration/manage_customers/).
+With this feature, your customers can [self-register](https://doc.ibexa.co/projects/userguide/en/latest/shop_administration/company_self_registration/),
+[edit their organization information](https://doc.ibexa.co/projects/userguide/en/latest/shop_administration/customer_portal/),
+invite and view members and check their order history.
+
+![Customer Portal dashboard](img/cp_dashboard_customer_portal.png)
+
+## Backend documentation
```
Comment (DominikaK) [inline docs/customer_management/customer_portal.md:15]: Not clear what this heading means, do you need it?

**Review by DominikaK — CHANGES_REQUESTED** [review]

File: docs/customer_management/create_user_registration_form.md (hunk 12 above)
Comment (MagdalenaZuba) [reply docs/customer_management/create_user_registration_form.md:10]: So I guess PhpStorm lied to me :D I will remove them

Comment (DominikaK) [reply docs/customer_management/create_user_registration_form.md:10]: Yes, Storm unfortunately does not understand autolinks, but autolinks are smarter ;)

**Review by DominikaK — APPROVED** [review]
